### PR TITLE
Rename Microsoft.Build.Tasks.Roslyn to Microsoft.Build.Tasks.CodeAnalysis

### PR DESCRIPTION
**Pull request by agocke**
@jaredpar @gafter @VSadov @AlekseyTs

### CODE REVIEW

File: src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class CompilerServerUnitTests : TestBase
         private const string CompilerServerExeName = "VBCSCompiler.exe";
         private const string CSharpClientExeName = "csc2.exe";
         private const string BasicClientExeName = "vbc2.exe";
-        private const string BuildTaskDllName = "Microsoft.Build.Tasks.Roslyn.dll";
+        private const string BuildTaskDllName = "Microsoft.Build.Tasks.CodeAnalysis.dll";
```
Comment (jaredpar) [inline src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs]: Hmm, seems like we need `assembly_nameof` feature  ;)

**Comment by jaredpar** [comment]
:+1:

**Comment by agocke** [comment]
ping -- can I get another sign off? Simple change

**Comment by agocke** [comment]
@jasonmalinowski As well

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**